core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -504,6 +506,13 @@ class KafkaServer(
           new FetchSessionCache(config.maxIncrementalFetchSessionCacheSlots,
             KafkaServer.MIN_INCREMENTAL_FETCH_SESSION_EVICTION_MS))
 
+        remoteLogManager.foreach(rlm => {
+          val listenerName = ListenerName.normalised(remoteLogManagerConfig.remoteLogMetadataManagerListenerName())
+          val endpoint = brokerInfo.broker.endPoints.find(e => e.listenerName.equals(listenerName))
+            .orElse(Some(brokerInfo.broker.endPoints.head))
```

Comment:
Yes, we need this endpoint for rlmm to work. I agree this should be restricted in the config level to avoid unexpected result in rlmm. Let me update it.